Posts/2021/03/File-System-Watcher-Engine-Event.md

```diff
@@ -1,42 +1,54 @@
+This includes:
 
-The PowerShell module
-[FSWatcherEngineEvent](https://www.powershellgallery.com/packages/FSWatcherEngineEvent) might help
-you out. The module provides commands for creating and managing file system watchers and will
-publish its notifications as PowerShell engine events instead of directly registered object events.
+- NotifyFilter: what kind of change triggers an event (by default: LastWrite, FileName, DirectoryName)
+- Filter: a wildcard to define a subset of files to watch
+- IncludeSubdirectory: extends the area of observation to the subdirectories of the specified path
 
-After installing and importing of the module a new filesystem watcher can easily be created with the
-'New-FileSystemWatcher' command. As an example it will watch for changes in directory
-'C:\temp\files'. The command allows to specify the same parameters (with the same names) as if you
-are using a FileSystemWatcher directly. This includes the notification filters, file name filter and
-option to include subdirectories. Please refer to the Microsofts reference documentation of the
-FileSystemWatcher class for the details.
+Please refer to the Microsofts reference documentation of the FileSystemWatcher class for the details.
 
 ```powershell
-PS> New-FileSystemWatcher -SourceIdentifier "myevent" -Path c:\temp\files
+New-FileSystemWatcher -SourceIdentifier "MyEvent" -Path C:\Temp\files
 ```
 
-The filesystem watcher will now send notifications to PowerShells engine event queue using the
-source identifier "myevent". The event can be consumed by registering an event handler for the same
-source identifier. The following example just writes the whole event converted to JSON to the
-console:
+The watcher now sends notifications to PowerShells engine event queue using the source identifier "MyEvent".
+You can consume the event by registering an event handler for the same source identifier.
+The following example just writes the whole event converted to JSON to the console:
 
-```powershell
-PS> Register-EngineEvent -SourceIdentifier "myevent" -Action { $event | ConvertTo-Json | Write-Host }
+```powershell-console
+PS> Register-EngineEvent -SourceIdentifier "MyEvent" -Action { $event | ConvertTo-Json | Write-Host }
 
 Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
 --     ----            -------------   -----         -----------     --------             -------
-1      myevent                         NotStarted    False                                $event|ConvertTo-Json|Wr…
+1      MyEvent                         NotStarted    False                                $event|ConvertTo-Json|Wr…
 ```
 
-To produce a new event some characters will be written to a file in the watched directory:
+PowerShell allows to register more than one handler for a single source identifier but the  FSWatcherEngineEvent module doesn't allow to create more than one watcher using the same source identifier.
 
```

Comment:
PowerShell allows you to register more than one handler for a single source identifier but the FSWatcherEngineEvent module doesn't allow you to create more than one watcher using the same source identifier.

Reply:
fixed